### PR TITLE
feat: handle empty names

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
- fix: default to name = '' not 'place holder'
- feat: add a welcome message
- feat: provide fallback when name is missing

- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

@Sboonny if you can think of any ways to make this look/feel a little nicer, that would be awesome.

I think this is a little nicer than the current approach (defaulting to calling everyone 'place holder'), but it's still a little rough.

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
One thing that this PR highlighted was that links are styled exactly the same as ordinary text.  Can we globally style them so they're distinct (and clearly links)?

**Comment by Sboonny** [comment]
about the links, sure in a second PR. I haven't manipulated the fonts because it's in roadmap, but I should have done the basics.

Give me a second, I will mess with the heading and code 👍

**Comment by Sboonny** [comment]
nah, I am create for the links. give me a hour, and will post review

File: client/src/modules/home/index.tsx [hunk 1]
```diff
@@ -5,19 +5,45 @@ import {
   GridItem,
   Button,
   useToast,
+  Center,
+  Box,
```
Comment (Sboonny) [inline client/src/modules/home/index.tsx:9]:
```suggestion
  Flex,
  Text,
```

File: client/src/modules/home/index.tsx [hunk 2]
```diff
@@ -5,19 +5,45 @@ import {
...
+      </Heading>
+      {!user.name && (
+        <Center>
+          <Box>
+            (you can set your name on your{' '}
+            <Link href="/profile">profile page</Link>)
+          </Box>
+        </Center>
+      )}
+    </>
```
Comment (Sboonny) [inline client/src/modules/home/index.tsx:38]:
```suggestion
  const helloUser = `Hello, ${user.name || 'annoymous'}`;
  return (
    <>
    <Flex alignItems={'center'} justifyContent='space-between'>
      <Heading as='h1'>
         {helloUser}
      </Heading>
      {!user.name && (
          <Text>
            You can set your name on your{' '}
            <Link href="/profile" textDecoration={"underline"} _hover={{textDecoration: 'none'}}>profile page</Link>
          </Text>
      )}
      </Flex>
```

File: client/src/modules/home/index.tsx [hunk 3]
```diff
@@ -40,29 +66,32 @@ const Home = () => {
...
+            {hasMore ? (
+              <Button onClick={onLoadMore}>Click for more</Button>
+            ) : (
+              <Heading size="md">No more</Heading>
+            )}
+          </VStack>
+        </GridItem>
+        <GridItem colSpan={{ base: 2, xl: 1 }}>
+          <VStack align="flex-start">
+            <Heading>Chapters</Heading>
```
Comment (Sboonny) [inline client/src/modules/home/index.tsx:87]:
```suggestion
      {user ?
       <Welcome user={user} />:
       <Heading as='h1'>Welcome to Chapter</Heading>
       }
      <Grid templateColumns="repeat(2, 1fr)" gap={10} mt="5">
        <GridItem colSpan={{ base: 2, xl: 1 }}>
          <VStack align="flex-start">
            <Heading as='h2' size={'md'}>Upcoming events</Heading>
            {data.paginatedEvents.map((event) => (
              <EventCard key={event.id} event={event} />
            ))}
            {hasMore ? (
              <Button onClick={onLoadMore}>Click for more</Button>
            ) : (
              <Text size="md">No more</Text>
            )}
          </VStack>
        </GridItem>
        <GridItem colSpan={{ base: 2, xl: 1 }}>
          <VStack align="flex-start">
            <Heading as='h2' size={'md'}>Chapters</Heading>
 ```

**Review by Sboonny — COMMENTED** [review]
I am surprised, that test aren't complaining. I am getting this error, and need to stop the server and rerun to please that error.

image

---

Here how it will look, while logged out 

image

----

Here how it looks, as user with a user name

image


---

here how it looks without a user name

image

**Comment by Sboonny** [comment]
I will make a link PR, and add margin top in the Polish chapter PR, feel free to merge this 👍.

File: client/src/modules/home/index.tsx [hunk 4]
```diff
@@ -5,19 +5,43 @@ import {
...
+  const helloUser = `Hello, ${user.name || 'annoymous'}`;
+  return (
+    <>
+    <Flex alignItems={'center'} justifyContent='space-between'>
+      <Heading as='h1'>
+         {helloUser}
+      </Heading>
+      {!user.name && (
+          <Text>
+            You can set your name on your{' '}
```
Comment (Sboonny) [inline client/src/modules/home/index.tsx:32]: does eslint hate this?

File: client/src/modules/home/index.tsx [hunk 5]
```diff
@@ -5,19 +5,43 @@ import {
...
 import { EventCard } from 'components/EventCard';
 import { useHomeQuery } from 'generated/graphql';
+import { AuthContextType, useAuth } from 'modules/auth/store';
 
+type User = NonNullable<AuthContextType['user']>;
+
+const Welcome = ({ user }: { user: User }) => {
+  const helloUser = `Hello, ${user.name || 'annoymous'}`;
+  return (
+    <>
```
Comment (Sboonny) [inline client/src/modules/home/index.tsx:25]:
```suggestion
```

Sorry about that

**Review by Sboonny — APPROVED** [review]

File: client/src/modules/home/index.tsx (hunk 5 above)
Comment (ojeytonwilliams) [reply client/src/modules/home/index.tsx]: No worries, it's easily done.

If there are a lot of changes, it's a pain to submit them through the GitHub UI. So, feel free to push a commit when that will save you time.

**Comment by Sboonny** [comment]
I am merging it, so it will be live

**Pull request merged**